### PR TITLE
fix: metrics endpoint must be secured behind authN

### PR DESCRIPTION
**Pull request by adodon2go**
**What type of PR is this?**
bug

**Which issue does this PR fix**:
https://github.com/project-zot/zot/issues/1859

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by peusebiu — DISMISSED** [review]
lgtm

File: pkg/api/errors/errors.go [hunk 1]
```diff
@@ -136,7 +136,7 @@ func NewError(code ErrorCode) *Error {
 
 		UNAUTHORIZED: {
 			Message: "authentication required",
-			Description: "The access controller was unable to authenticate the client." +
+			Description: "The access controller was unable to authenticate the client. " +
 				"Often this will be accompanied by a Www-Authenticate HTTP response header " +
```
Comment (rchincha) [inline pkg/api/errors/errors.go:140]: nitpick ... "WWW-Authenticate" let's fix this as we catch them.

File: test/blackbox/metrics.bats [hunk 2]
```diff
@@ -52,11 +64,35 @@ function setup_file() {
...
+        "address": "0.0.0.0",
+        "port": "9000",
+        "auth": {
+            "htpasswd": {
+                "path": "${zot_htpasswd_file}"
+            }
+        }
+    },
+    "log": {
+        "level": "debug"
```
Comment (andaaron) [inline test/blackbox/metrics.bats:85]: Maybe add a log file and print the output on failure in teardown()

File: test/blackbox/metrics.bats [hunk 3]
```diff
@@ -19,14 +25,14 @@ function setup_file() {
         exit 1
     fi
 
-    # Download test data to folder common for the entire suite, not just this file
-    skopeo --insecure-policy copy --format=oci docker://ghcr.io/project-zot/golang:1.20 oci:${TEST_DATA_DIR}/golang:1.20
```
Comment (andaaron) [inline test/blackbox/metrics.bats:23]: Is no longer necessary?

**Review by andaaron — APPROVED** [review]

**Pull request merged**